cluster-autoscaler/main.go

```diff
@@ -205,22 +205,23 @@ var (
+	recordDuplicatedEvents                    = flag.Bool("record-duplicated-events", false, "enable duplication of similar events within a 5 minute window.")
+	maxNodesPerScaleUp                        = flag.Int("max-nodes-per-scaleup", 1000, "Max nodes added in a single scale-up. This is intended strictly for optimizing CA algorithm latency and not a tool to rate-limit scale-up throughput.")
+	maxNodeGroupBinpackingDuration            = flag.Duration("max-nodegroup-binpacking-duration", 10*time.Second, "Maximum time that will be spent in binpacking simulation for each NodeGroup.")
+	skipNodesWithSystemPods                   = flag.Bool("skip-nodes-with-system-pods", true, "If true cluster autoscaler will never delete nodes with pods from kube-system (except for DaemonSet or mirror pods)")
+	skipNodesWithLocalStorage                 = flag.Bool("skip-nodes-with-local-storage", true, "If true cluster autoscaler will never delete nodes with pods with local storage, e.g. EmptyDir or HostPath")
+	minReplicaCount                           = flag.Int("min-replica-count", 0, "Minimum number or replicas that a replica set or replication controller should have to allow their pods deletion in scale down")
+	nodeDeleteDelayAfterTaint                 = flag.Duration("node-delete-delay-after-taint", 5*time.Second, "How long to wait before deleting a node after tainting it")
+	scaleDownSimulationTimeout                = flag.Duration("scale-down-simulation-timeout", 5*time.Minute, "How long should we run scale down simulation.")
+	parallelDrain                             = flag.Bool("parallel-drain", false, "Whether to allow parallel drain of nodes.")
+	maxCapacityMemoryDifferenceRatio          = flag.Float64("memory-difference-ratio", config.DefaultMaxCapacityMemoryDifferenceRatio, "Maximum difference in memory capacity between two similar node groups to be considered for balancing. Value is a ratio of the smaller node group's memory capacity.")
+	maxFreeDifferenceRatio                    = flag.Float64("max-free-difference-ratio", config.DefaultMaxFreeDifferenceRatio, "Maximum difference in free resources between two similar node groups to be considered for balancing. Value is a ratio of the smaller node group's free resource.")
+	maxAllocatableDifferenceRatio             = flag.Float64("max-allocatable-difference-ratio", config.DefaultMaxAllocatableDifferenceRatio, "Maximum difference in allocatable resources between two similar node groups to be considered for balancing. Value is a ratio of the smaller node group's allocatable resource.")
+	forceDaemonSets                           = flag.Bool("force-ds", false, "Blocks scale-up of node groups too small for all suitable Daemon Sets pods.")
+	allowScaleDownOnCustomControllerOwnedPods = flag.Bool("allow-scale-down-on-custom-controller-owned-pods", false, "Don't block node scale-down if a pod owned by a custom controller is running on the node.")
```

Comment:
Better naming suggestions are welcome (the flag name feels a little too long).

Reply:
The default is set to `false` right now to provide backwards compatibility. This flag would be set to `true` in the future (we might remove it completely and use `true` one as the default behavior).

Reply:
@vadasambar Just an idea, maybe you could follow the `skip-nodes-with-*` scheme, for instance,
`skip-nodes-with-custom-controller-pods`?

Reply:
@gregth thank you for the suggestion. I thought the new flag name `skip-nodes-with-custom-controller-pods` wouldn't tell the user about if the skipping happens during scale-up or scale-down but looks like we already use `skip-nodes-with-*` flag naming for skipping things when scaling down nodes:
https://github.com/kubernetes/autoscaler/blob/1acb6b23608c50cdec80c2846c690c82e51377d0/cluster-autoscaler/main.go#L214-L215

Making the naming consistent makes sense to me. I will change it to `skip-nodes-with-custom-controller-pods`. Thank you.

P.S.: If you have any other suggestions, I would love to know.